### PR TITLE
Add @pixi/webworker bundle

### PR DESCRIPTION
**Pull request by SuperSodaSea**
This PR adds `@pixi/webworker` bundle for Web Workers. This bundle is basicly `pixijs` with `@pixi/accessibility` & `@pixi/events` removed and `WebWorkerAdapter` added, enough for running small demos in Web Workers.

See also https://github.com/pixijs/examples/pull/158.

Related Issue: https://github.com/pixijs/pixijs/issues/7123

##### Description of change

- Add `@pixi/webworker` bundle
- Add `IAdapter.getFontFaceSet()` to cover the difference between main thread and worker thread (`document.fonts` for main thread, `globalThis.fonts` for worker thread)
- `ImageBitmapResource` now recieve `string` in constructor, in order to support `Texture.from(url)` in Web Workers
- Other minor changes to make basic things work in Web Workers

##### Todo List

- [x] (Fixed) `Texture.from(url)` not working (It use `ImageResource`, which is not working since HTMLImageElement is not available in Web Workers. Perhaps we can make `ImageBitmapResource` recieve `string`). Using `await Assets.load(url)` is OK.

##### Pre-Merge Checklist

- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)

### CODE REVIEW

**Comment by andrewstart** [comment]
I don't really have the time to test this myself, but does this work in Firefox? Does [`gl.commit()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/commit) need to be used (where available)?

**Comment by SuperSodaSea** [comment]
> I don't really have the time to test this myself, but does this work in Firefox? Does [`gl.commit()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/commit) need to be used (where available)?

@andrewstart The example works just fine in my Firefox 105.
Firefox 105

The HTML Standard doesn't describe `WebGLRenderingContext.commit()` clearly, but it should be similar to [`OffscreenCanvasRenderingContext2D.commit()`](https://html.spec.whatwg.org/multipage/canvas.html#offscreencontext2d-commit):

> **offscreenCanvasRenderingContext2D.[commit](https://html.spec.whatwg.org/multipage/canvas.html#offscreencontext2d-commit)()**
> Copies the rendering context's [bitmap](https://html.spec.whatwg.org/multipage/canvas.html#offscreencontext2d-bitmap) to the bitmap of the [placeholder canvas element](https://html.spec.whatwg.org/multipage/canvas.html#offscreencanvas-placeholder) of the [associated OffscreenCanvas object](https://html.spec.whatwg.org/multipage/canvas.html#associated-offscreencanvas-object). The copy operation is synchronous. Calling this method is not needed for the transfer, since it happens automatically during the [event loop](https://html.spec.whatwg.org/multipage/webappapis.html#event-loop) execution.

It seems that `commit()` is used to force the browser to copy the content from OffscreenCanvas to original canvas (synchronously), which is automatically done by the event loop. That is to say if we use `PIXI.Application`'s ticker (just like how we do in the examples) or something like `setTimeout` or `requestAnimationFrame` to render, the `commit()` is not necessary. It is only necessary to call `commit()` if we manually write a loop and keep rendering without going back to the event loop, for example:

```js
// These codes are running in worker thread

// Won't work, never go back to event loop and nothing displayed
while (true) {
    container.rotation -= 0.01;
    renderer.render();
}

// Work for both Chrome and FireFox, go back to event loop and copy canvas content when wait()
function wait() { return new Promise(resolve => setTimeout(resolve, 0)); }
while (true) {
    container.rotation -= 0.01;
    renderer.render();
    await wait();
}

// Work for Firefox (Chrome don't have gl.commit() currently)
while (true) {
    container.rotation -= 0.01;
    renderer.render();
    renderer.gl.commit();
}
```

**TL;DR**: The `commit()` is not necessary in most the cases. I can still add a method like `IRenderer.commit()` if wanted.

**Comment by andrewstart** [comment]
Thanks for looking into that - this is actually the first I had heard of the feature, and given how recently Firefox added support I just wanted to make sure there wasn't a (purely hypothetical) situation where it worked in Chrome with an older version of the WIP spec but not Firefox with a later version. No changes needed from me.

File: packages/core/src/textures/resources/ImageBitmapResource.ts [hunk 1]
```diff
@@ -1,27 +1,196 @@
...
+    private static __DUMMY: ICanvas;
+
+    private static get _DUMMY(): ICanvas
+    {
+        if (ImageBitmapResource.__DUMMY) return ImageBitmapResource.__DUMMY;
+
+        ImageBitmapResource.__DUMMY = settings.ADAPTER.createCanvas(0, 0);
+
+        return ImageBitmapResource.__DUMMY;
     }
```
Comment (bigtimebuddy) [inline packages/core/src/textures/resources/ImageBitmapResource.ts:195]: "Dummy" naming could be improved :) Also comments are helpful. 

```suggestion
    /** Temporary empty canvas to use when loading URLs */
    private static _EMPTY: ICanvas;
   
    /** Get cached temp canvas */
    private static get EMPTY(): ICanvas
    {
        ImageBitmapResource._EMPTY = ImageBitmapResource._EMPTY ?? settings.ADAPTER.createCanvas(0, 0);

        return ImageBitmapResource._EMPTY;
    }
```

Comment (SuperSodaSea) [reply packages/core/src/textures/resources/ImageBitmapResource.ts]: Done in ca8527757166676cf4d8e4482d072bd513469093.

File: bundles/pixi.js-webworker/src/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { settings } from '@pixi/settings';
+
+import type { IAdapter } from '@pixi/settings';
```
Comment (bigtimebuddy) [inline bundles/pixi.js-webworker/src/adapter.ts:3]: We should import from core instead. settings/constants/ticker/utils/runner are all internal packages only to core.

```suggestion
import { settings } from '@pixi/core';

import type { IAdapter } from '@pixi/core';
```

File: packages/core/src/textures/resources/ImageBitmapResource.ts [hunk 3]
```diff
@@ -1,27 +1,203 @@
+import { ALPHA_MODES } from '@pixi/constants';
+import { settings } from '@pixi/settings';
```
Comment (bigtimebuddy) [inline packages/core/src/textures/resources/ImageBitmapResource.ts:2]:
```suggestion
import { ALPHA_MODES, settings } from '@pixi/core';
```

File: packages/core/src/textures/resources/ImageResource.ts [hunk 4]
```diff
@@ -1,14 +1,14 @@
-import { BaseImageResource } from './BaseImageResource';
-import { settings } from '@pixi/settings';
 import { ALPHA_MODES } from '@pixi/constants';
+import { settings } from '@pixi/settings';
```
Comment (bigtimebuddy) [inline packages/core/src/textures/resources/ImageResource.ts:2]:
```suggestion
import { ALPHA_MODES, settings } from '@pixi/core';
```

**Review by bigtimebuddy — COMMENTED** [review]
Need to sort some imports, but otherwise looks great.

File: packages/core/src/textures/resources/ImageBitmapResource.ts (hunk 3 above)
Comment (SuperSodaSea) [reply packages/core/src/textures/resources/ImageBitmapResource.ts]: `ImageResource` and `ImageBitmapResource` are inside of `@pixi/core`, they can't import `@pixi/core` itself.

Comment (bigtimebuddy) [reply packages/core/src/textures/resources/ImageBitmapResource.ts:2]: Yeah, my bad!

**Review by bigtimebuddy — APPROVED** [review]

**Comment by bigtimebuddy** [comment]
Thanks @SuperSodaSea! Another great PR
